akka-docs/src/main/paradox/typed/style-guide.md

```diff
@@ -457,6 +457,29 @@ of using `newReceiveBuilder`. Implement the `receiveMessage` and `receiveSignal`
 
 @@@
 
+## Nesting setup
+
+When an actor behavior needs more than one of `setup`, `withTimers` and `withStash` the methods can be nested to access
+the needed dependencies:
+
+Scala
+:  @@snip [StyleGuideDocExamples.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/StyleGuideDocExamples.scala) { #nesting }
+
+Java
+:  @@snip [StyleGuideDocExamples.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/StyleGuideDocExamples.java) { #nesting }
+
+The order of the nesting does not change the behavior as long as there is no additional logic in any other function than the innermost one. 
+It can be nice to default to put `setup` outermost as that is the least likely block that will be removed if the actor logic changes. 
```

Comment:
Not sure we have any other recommendations here? I like it because it is often conceptually "preStart" as well, but otoh only using `withStash { stash => do stuff; behavior }` also ends up being a pre-start.

Reply:
Maybe, but OTOH if nesting you'd want to put the logic innermost and just use the outer lambdas to close over the dependencies.